compiler/rustc_interface/src/interface.rs

```diff
@@ -183,13 +184,16 @@ pub fn parse_check_cfg(specs: Vec<String>) -> CheckCfg {
                                 if let Some((name, values)) = args.split_first() {
                                     if name.is_word() && name.ident().is_some() {
                                         let ident = name.ident().expect("multi-segment cfg key");
-                                        cfg.values_checked.insert(ident.to_string());
+                                        let ident_values = cfg
+                                            .values_valid
+                                            .entry(ident.to_string())
```

Comment:
`ident.to_string()` is too smart and is supposed to only be used in diagnostics (see how `impl fmt::Display for Ident` is implemented).
`ident.name`/`ident.name.to_string()` needs to be used here instead.